### PR TITLE
Add gotm test

### PR DESCRIPTION
**Pull request by mark-petersen**
Moved directly from https://github.com/MPAS-Dev/MPAS-Model/pull/704

### CODE REVIEW

**Review by qingli411 — APPROVED** [review]
I just tested it and it works.

File: ocean/periodic_planar/2.5km/default_gotm/config_forward.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		<option name="config_ocean_run_mode">'forward'</option>
+		<option name="config_dt">'000:00:25'</option>
+		<option name="config_btr_dt">'000:00:25'</option>
+		<option name="config_time_integrator">'split_explicit'</option>
+		<option name="config_run_duration">'0000_12:00:00'</option>
+		<option name="config_zonal_ssh_grad">-1.0e-5</option>
+		<option name="config_pressure_gradient_type">'constant_forced'</option>
+		<option name="config_use_cvmix">.false.</option>
+		<option name="config_use_gotm">.true.</option>
+        <option name="config_gotm_namelist_file">'gotmturb.nml'</option>
```
Comment (xylar) [inline ocean/periodic_planar/2.5km/default_gotm/config_forward.xml:20]: Indentation here needs to be fixed

**Comment by xylar** [comment]
@mark-petersen and @qingli411, this would be an easy test case to port to the new python package once #28 gets merged, hopefully by the end of the month.  Unless either of you wants to take that on, I'll be happy to take care of it.

**Comment by qingli411** [comment]
Thanks, @xylar! I think it would be better if you could help with this. I'm still trying to get myself familiar with the new python package.

**Comment by qingli411** [comment]
@xylar, sure, that would be great! Looking forward to the tutorial!

**Comment by xylar** [comment]
@mark-petersen, could you update the `MPAS-Model/ocean/develop` submodule now that MPAS-Dev/MPAS-Model#704 has been merged?  Then, I'll confirm that I an run the tests and merge assuming I can.

File: ocean/periodic_planar/2.5km/default_gotm/config_forward.xml (hunk 1 above)
Comment (xylar) [reply ocean/periodic_planar/2.5km/default_gotm/config_forward.xml:20]: @mark-petersen, please also fix this indentation issue if you don't mind.

File: ocean/periodic_planar/2.5km/default_gotm/config_analysis.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<config case="analysis">
+	<add_link source="../forward/output.nc" dest="output.nc"/>
+	<add_link source="../plot_profile.py" dest="plot_profile.py"/>
+	<copy_file source_path="script_test_dir" source="plot_profile.py" dest="plot_profile.py"/>
```
Comment (xylar) [inline ocean/periodic_planar/2.5km/default_gotm/config_analysis.xml:5]: I think this should be linked only, not copied and it should be linked inside the case directory, not the test directory.

Comment (xylar) [reply ocean/periodic_planar/2.5km/default_gotm/config_analysis.xml:5]:
```suggestion
	<add_link source_path="script_test_dir" source="plot_profile.py" dest="plot_profile.py"/>
```

**Review by xylar — COMMENTED** [review]
I don't know how much it matters because we're going to move away from `legacy` soon but I would put this in a `gotm` configuration instead of `periodic_planar` (which is too generic a configuration name to be useful).  Then, I would just call the test `default`.

**Comment by xylar** [comment]
@qingli411, @mark-petersen is out right now.  If you would like to get this work merged, the easiest way might be to make a copy of @mark-petersen's branch and then make the modifications I requested.  Or we can wait until @mark-petersen gets back.

**Comment by qingli411** [comment]
@xylar OK I will do that.

**Comment by xylar** [comment]
closed via #86

**Pull request closed without merging**